src/veins/modules/mobility/traci/subscriptionManagement/PersonSubscriptionManager.h

```diff
@@ -0,0 +1,153 @@
+#include <string>
+#include <list>
+
+#include "veins/modules/mobility/traci/subscriptionManagement/SubscriptionManagerBase.h"
+#include "veins/modules/mobility/traci/subscriptionManagement/TraCIPerson.h"
+
+namespace veins {
+namespace TraCISubscriptionManagement {
+
+/**
+ * @class PersonSubscriptionManager
+ *
+ * Manages all person subscriptions. You can update it via a subscription
+ * result in the form of a TraCIBuffer or via a manual update but it is
+ * necessary to provide an id_list of active persons then.
+ *
+ * After updating you can request the disappeared and appeared persons via
+ * the class methods.
+ *
+ * IMPORTANT: Call initialize() before using!
+ *
+ * @author Nico Dassler <dassler@hm.edu>
+ */
+class PersonSubscriptionManager: public SubscriptionManagerBase {
+
+public:
+    /**
+     * Constructor.
+     */
+    PersonSubscriptionManager();
+
+    /**
+     * Default destructor.
+     */
+    virtual ~PersonSubscriptionManager() = default;
+
+    /**
+     * Update this PersonSubscriptionManager with the given
+     * active person id list. You can force an update of active
+     * persons using this.
+     *
+     * @param currentlyActivePersonIds A list containing the currently active persons
+     * identified by their ids.
+     *
+     * @return true
+     */
+    bool updateWithList(std::list<std::string>& currentlyActivePersonIds);
+
+    /**
+     * Update this manager with the given buffer containing a subscription
+     * response for a person value.
+     *
+     * This will create new subscriptions for specific persons if the buffer
+     * contained an ID_LIST subscription.
+     *
+     * @param buffer the traci buffer containing the subscription response.
+     *
+     * @return bool true if the given buffer contained an ID_LIST response.
+     */
+    bool update(TraCIBuffer& buffer) override;
+
+    /**
+     * This gives you all the persons that were updated (includes
+     * new persons) since the last time you called this method. That
+     * means that multiple calls to this method will NOT yield the same
+     * result.
+     *
+     * This can also contain multiple updates for a single person.
+     *
+     * @return a list of TraCIPerson.
+     */
+    std::list<TraCIPerson> getUpdated();
+
+    /**
+     * This gives you all the disappeared persons since you called
+     * this method the last time. That means that multiple calls
+     * to this method will yield different results.
+     *
+     * @return std::set<std::string> a set of person ids.
+     */
+    std::set<std::string> getDisappeared();
+
+    /**
+     * Initialize this manager with the given parameters to access TraCI.
+     */
+    void initialize(std::shared_ptr<TraCIConnection> connection, std::shared_ptr<TraCICommandInterface> commandInterface) override;
+
+private:
+
+    /**
+     * This contains all updated persons. Will be cleared after
+     * getUpdated() was called.
+     */
+    std::list<TraCIPerson> mUpdatedPersons;
```

Comment:
Any particular (and good) reason to use a std::list here? If not,just us a std::vector. (As above)

Reply:
modified.

Reply:
Alright.